crates/matrix-sdk-ui/src/timeline/read_receipts.rs

```diff
@@ -279,11 +396,53 @@ impl TimelineInnerStateTransaction<'_> {
 
         self.meta.read_receipts.maybe_update_read_receipt(
             full_receipt,
-            None,
             is_own_event,
+            &self.meta.all_events,
             &mut self.items,
         );
     }
+
+    /// Update the read receipts on the event with the given event ID and the
+    /// previous visible event because of a visibility change.
+    pub(super) fn maybe_update_read_receipts_of_prev_event(&mut self, event_id: &EventId) {
+        // Find the previous visible event, if there is one.
+        let Some((prev_item_pos, prev_event_item)) = self
+            .all_events
+            .iter()
+            .rev()
+            // Find the event item.
+            .skip_while(|meta| meta.event_id != event_id)
+            // Go past the event item.
+            .skip(1)
+            // Find the first visible item.
+            .find(|meta| meta.visible)
+            .and_then(|meta| rfind_event_by_id(&self.items, &meta.event_id))
```

Comment:
Should the `find` and `rfind_event_by_id` be separate? I imagine the first one returning `None` is not exceptional, but the second one returning `None` after the first one returned `Some(event_id)` would mean some sort of internal inconsistency (that should be logged)?

Reply:
Done